### PR TITLE
chore: update source for aws samples input data

### PR DESCRIPTION
**Pull request by ryansteakley**
**Description of your changes:**

- old deeplearning link is now unavailable.
- Get new source from aws-samples-repo for kmeans jobs

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md [hunk 1]
```diff
@@ -67,9 +67,18 @@ def np2csv(arr):
...
-    train_set, _, _ = pickle.load(f, encoding='latin1')
+# Load the dataset
+s3 = boto3.client("s3")
+data_bucket = S3DataConfig(
+    sm_session, "example-notebooks-data-config", "config/data_config.json"
+).get_data_bucket()
+print(f"Using data from {data_bucket}")
+
+s3.download_file(
+    data_bucket, "datasets/image/MNIST/mnist.pkl.gz", "mnist.pkl.gz")
```
Comment (akartsky) [inline samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md:78]: how big is this file ? 
make sure it is not too big for example script

**Review by akartsky — APPROVED** [review]

File: samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md (hunk 1 above)
Comment (ryansteakley) [reply samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md:78]: the input data is around ~28mb. The test_data and train_data are each around 150-180mb which is similiar to that of the sample data used in our ack sagemaker repo

File: samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md [hunk 2]
```diff
@@ -67,9 +67,18 @@ def np2csv(arr):
     return csv.getvalue().decode().rstrip()
 
 # Prepare input for the model
-urllib.request.urlretrieve("http://deeplearning.net/data/mnist/mnist.pkl.gz", "mnist.pkl.gz")
-with gzip.open('mnist.pkl.gz', 'rb') as f:
-    train_set, _, _ = pickle.load(f, encoding='latin1')
+# Load the dataset
+s3 = boto3.client("s3")
+data_bucket = S3DataConfig(
+    sm_session, "example-notebooks-data-config", "config/data_config.json"
```
Comment (surajkota) [inline samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md:73]: undefined var `sm_session`? and dont we need to import `S3DataConfig`?

Comment (ryansteakley) [reply samples/contrib/aws-samples/mnist-kmeans-sagemaker/README.md:73]: yes you are correct will amend these files.

**Pull request merged**